dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphUtility/BarGraph.cs

```diff
@@ -0,0 +1,479 @@
+            renderTarget.DrawText(xAxisText, textFormatHorizontal, ClientRectangleXAxis, textSceneColorBrush);
+
+            renderTarget.EndDraw();
+
+            // Rotate render target to draw y axis text.
+            renderTarget.Transform = Matrix3x2.Rotation((float)(-Math.PI / 2), new Vector2(0, containerHeight));
+
+            renderTarget.BeginDraw();
+
+            renderTarget.DrawText(yAxisText, textFormatVertical, ClientRectangleYAxis, textSceneColorBrush);
+
+            renderTarget.EndDraw();
+
+            // Rotate the RenderTarget back.
+            renderTarget.Transform = Matrix3x2.Identity;
+        }
+
+        //Dispose of resources.
+        public void Dispose()
+        {
+            textSceneColorBrush.Dispose();
+            textFormatTitle.Dispose();
+            textFormatHorizontal.Dispose();
+            textFormatVertical.Dispose();
+        }
+
+        private Bar[] CreateBars(float[] data)
+        {
+            //Clear hashmap.
+            barValueMap.Clear();
+
+            var barBrushHelper = new BarGraphUtility.BarBrushHelper(compositor);
+            var brushes = new CompositionBrush[data.Length];
+            CompositionBrush brush = null;
+
+            switch (graphBarStyle)
+            {
+                case GraphBarStyle.Single:
+                    brush = barBrushHelper.GenerateSingleColorBrush(graphBarColors[0]);
+                    break;
+                case GraphBarStyle.Random:
+                    brushes = barBrushHelper.GenerateRandomColorBrushes(data.Length);
+                    break;
+                case GraphBarStyle.PerBarLinearGradient:
+                    brush = barBrushHelper.GenerateLinearGradient(graphBarColors);
+                    break;
+                case GraphBarStyle.AmbientAnimatingPerBarLinearGradient:
+                    brush = barBrushHelper.GenerateAmbientAnimatingLinearGradient(graphBarColors);
+                    break;
+                default:
+                    brush = barBrushHelper.GenerateSingleColorBrush(graphBarColors[0]);
+                    break;
+            }
+
+            var maxValue = maxBarValue = GetMaxBarValue(data);
+            var bars = new Bar[data.Length];
+            for (int i = 0; i < data.Length; i++)
+            {
+                var xOffset = shapeGraphOffsetX + barSpacing + (barWidth + barSpacing) * i;
+                var height = GetAdjustedBarHeight(maxValue, graphData[i]);
+                var barBrush = brush ?? brushes[i];
+
+                var bar = new BarGraphUtility.Bar(compositor, shapeGraphContainerHeight, height, barWidth, "something", graphData[i], barBrush);
+                bar.OutlineRoot.Offset = new System.Numerics.Vector3(xOffset, shapeGraphContainerHeight, 0);
+                bar.Root.Offset = new System.Numerics.Vector3(xOffset, shapeGraphContainerHeight, 0);
+
+                barValueMap.Add(i, bar);
```

Comment:
Seeing as the key to the hashtable is an integer, and you're adding them in order, the data structure can just be an array - no need for a hashtable. You're already creating an array so you don't need barValueMap at all.

Reply:
updated